includes/admin/class-wc-rest-payments-payment-intents-controller.php

```diff
@@ -46,4 +56,35 @@ public function get_payment_intent( $request ) {
 
 		return $this->forward_request( 'get_intent', [ $payment_intent_id ] );
 	}
+
+	/**
+	 * Create a payment intent.
+	 *
+	 * @param WP_REST_Request $request data about the request.
+	 */
+	public function post_payment_intent( $request ) {
+		try {
+			$metadata = $request->get_param( 'metadata' );
```

Comment:
It seems that you're using the structure of PaymentIntent itself for this code. However, here I think we're going to receive a request to create a new payment intent so I think we just need to have `order_id` for requests. 

Note: `order_id` is really the post ID (int) while `order_number` can be a string in Woo. 

`metadata` and `level3` params are not necessary for this request. We should build up them from what we have already had (order_id and order object): 

https://github.com/Automattic/woocommerce-payments/blob/232acdfa2ce16abe1bee8efbd86e31e143909c5e/includes/class-wc-payment-gateway-wcpay.php#L1115-L1116

https://github.com/Automattic/woocommerce-payments/blob/232acdfa2ce16abe1bee8efbd86e31e143909c5e/includes/class-wc-payment-gateway-wcpay.php#L2918-L2921

Also, in terms of security, we should not allow to update the whole `metadata` of payment intent directly from REST. We have some critical functions (fraud, integrity checks before processing) rely on it.

Reply:
Fixed.